inductiva/utils/format_utils.py

```diff
@@ -40,55 +40,60 @@ def seconds_formatter(secs: float) -> str:
+    """Apply formatters to a list of lists of data (rows).
 
-    Temporary solution to display tables. `pandas` dependency to print tables
-    is overkill, but since we have `pandas` anyway for now, we can use it.
+    Args:
+        table_data : Dictionary of column names and lists of data.
+        formatters : Dictionary of column names and functions to
+            apply to that column's data.
     """
-    df = pd.DataFrame(rows, columns=columns)
-    formatters = formatters or {}
-
-    col_space = {col: default_col_space for col in columns}
-    col_space.update(override_col_space or {})
-
-    # replace None with np.nan so that pandas can format them as "n/a"
-    # by passing na_rep="n/a" to to_string()
-    df.fillna(np.nan, inplace=True)
-
-    return df.to_string(
-        index=False,
-        na_rep="n/a",
-        formatters=formatters,
-        col_space=col_space,
-    )
+    for column_name, formatter in formatters.items():
+        if column_name in table_data:
+            table_data[column_name] = [
+                formatter(x) for x in table_data[column_name]
+            ]
+
+    return table_data
+
+
+def get_tabular_str(tabular_data: Union[Mapping[str, Iterable[Any]],
+                                        Iterable[Iterable[Any]]],
+                    headers: list = (),
+                    formatters: dict = None) -> str:
+    """Converts the list rows to a string table.
+
+    Args:
+        tabular_data: can be a list-of-lists (or another iterable of 
+            iterables), a list of named tuples, a dictionary of
+            iterables, an iterable of dictionaries, an iterable of
+            dataclasses (Python 3.7+)
+        headers: A list of column names.
+            Only needed if tabular_data is not a Mapping.
+        formatters: A dictionary of column names and functions to apply
+            to the data in that column. The function should take a single
+            argument and return a string. The function will be applied to the
+            data in the column before printing. Defaults to None.
 
-
-def get_dataframe_str(dataframe,
-                      default_col_space=15,
-                      override_col_space=None,
-                      formatters=None):
-    """Converts a dataframe to a string table.
-
-    Temporary solution to display tables. `pandas` dependency to print tables
-    is overkill, but since we have `pandas` anyway for now, we can use it.
     """
+
     formatters = formatters or {}
 
-    col_space = {col: default_col_space for col in dataframe.columns}
-    col_space.update(override_col_space or {})
+    if not isinstance(tabular_data, Mapping):
+        data = {}
+
+        for index, column_name in enumerate(headers):
+            data[column_name] = [row[index] for row in tabular_data]
+        #if we have no headers data will be empty.
+        #So, we want our original tabular_data
+        if headers != ():
+            tabular_data = data
```

Comment:
I am a bit confused at the moment with this... In the comment you state if we have no headers data will be empty and we want our original tabular data. Doesn't this imply:

```suggestion
        #if we have no headers data will be empty.
        #So, we want our original tabular_data
        if headers == ():
            tabular_data = data
```